### ISSUE TITLE
2.0 eslint rule newline-per-chained-call bug with array access

### ISSUE THREAD

**Issue by evilebottnawi** [eslint/eslint#5245]
```javascript
var currentUrl = window
    .location
    .href
    .match(/(^[^#]*)/)[0];
```
Output:
```
1:18  error  Expected line break after `window\n    .location\n    .href\n    .match(/(^[^#]*)/)`
```

But:
```javascript
var currentUrl = window
    .location
    .href
    .match(/(^[^#]*)/);
```
Work fine.

**Comment by albertosantini** [comment]
just an idea to reproduce the issue:

```
17:45:43 [/c/My/Dev]
λ eslint --version
v2.0.0
17:45:55 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)" | eslint --stdin --rule='newline-per-chained-call: 2'
17:46:05 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)[0]" | eslint --stdin --rule='newline-per-chained-call: 2'

<text>
  1:1  error  Expected line break after `window\n.location\n.href\n.match(/(^[^#]*)/)`  newline-per-chained-call

✖ 1 problem (1 error, 0 warnings)

17:46:12 [/c/My/Dev]
λ
```

**Comment by albertosantini** [comment]
Be aware, using `ignoreChainWithDepth` (default equals to 2):
http://eslint.org/docs/rules/newline-per-chained-call


```
17:57:56 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)[0]" | eslint --stdin --rule='newline-per-chained-call: [2,{ignor
eChainWithDepth: 4} ]'

<text>
  1:1  error  Expected line break after `window\n.location\n.href\n.match(/(^[^#]*)/)`  newline-per-chained-call

✖ 1 problem (1 error, 0 warnings)

17:58:01 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)[0]" | eslint --stdin --rule='newline-per-chained-call: [2,{ignor
eChainWithDepth: 5} ]'
17:58:05 [/c/My/Dev]
λ
```

**Comment by evilebottnawi** [comment]
@albertosantini i think it is bug, array access should not considered in chain.

**Comment by ilyavolodin** [comment]
Sounds like bug, I agree that array access shouldn't be part of the check here.

**Comment by thedark1337** [comment]
Personally , I want to ask if its possible to have an option to chain function calls only and not object access as well?

 I really don't mind the newline-per-chained-call for those but it seems sometimes it gives me a false positive when it shouldn't. Such as what is reported here (Array access) and some weird ones such as `options.level` and `Object.keys(options.meta)`

Example:

```
   37:128  error    Expected line break after `Object.keys(options.meta)`                                          newline-per-chained-call
   37:128  error    Expected line break after `Object`                                                             newline-per-chained-call
   97:19   error    Expected line break after `b[c].split(';')`                                                    newline-per-chained-call
   97:19   error    Expected line break after `b[c]`                                                               newline-per-chained-call
   97:19   error    Expected line break after `b`
```

```
// line 37
                return `${options.timestamp()} ${level} ${(options.message != null ? options.message : '')} ${(options.meta && Object.keys(options.meta).length ? `
                    ${JSON.stringify(options.meta)}
                    ` : '')}`;

// line 97 
        const d = b[c].split(';')[0];
```



Using ESLint Version 2.0 with my [ESLint Config](https://github.com/thedark1337/eslint-config-thedark1337)

**Comment by BYK** [comment]
@thedark1337 - Agree with an option to ignore members that are not called but I think that should be a separate issue as an enhancement proposal.

**Comment by ilyavolodin** [comment]
@thedark1337 Could you open a new issue with the proposal for a new option please?
